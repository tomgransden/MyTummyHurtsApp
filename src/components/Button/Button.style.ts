import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    width: 250,
    minHeight: 48,
    alignSelf: 'center',
    backgroundColor: 'mediumpurple',
    borderWidth: 1,
    borderColor: 'mediumpurple',
    borderRadius: 4,
    padding: 8,
    marginBottom: 16,
  },
  buttonText: {
    fontFamily: 'RubikBubbles-Regular',
    fontSize: 20,
    color: 'white',
    textAlign: 'center',
  },
});

export default styles;
