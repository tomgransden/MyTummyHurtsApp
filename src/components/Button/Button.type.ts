export interface IButtonProps {
  label: string;
  onPress: () => void;
  loading?: boolean;
}
