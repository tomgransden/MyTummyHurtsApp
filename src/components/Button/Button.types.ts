import { TextStyle, ViewStyle } from 'react-native';

export type ButtonProps = {
  title: string;
  onPress: () => void;
};

export type ButtonStyle = {
  container: ViewStyle;
  buttonText: TextStyle;
};
