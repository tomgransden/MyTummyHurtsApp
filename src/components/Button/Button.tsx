import React from 'react';
import { TouchableOpacity, Text, ActivityIndicator } from 'react-native';

import styles from './Button.style';
import { IButtonProps } from './Button.type';

const Button = ({ title, onPress, loading }: IButtonProps) => (
  <TouchableOpacity style={styles.container} onPress={loading ? undefined : onPress}>
    {loading ? <ActivityIndicator color="white" /> : <Text style={styles.buttonText}>{title}</Text>}
  </TouchableOpacity>
);

export default Button;
