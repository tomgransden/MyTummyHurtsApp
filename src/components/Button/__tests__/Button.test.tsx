import { fireEvent, render, screen } from '@testing-library/react-native';

import Button from '../Button';

describe('Button', () => {
  it('should render correctly', () => {
    const onPress = jest.fn();
    render(<Button title="My mock button" onPress={onPress} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should call onPress', () => {
    const onPress = jest.fn();
    render(<Button onPress={onPress} title="My mock button" />);
    screen.debug();
    fireEvent.press(screen.getByText('My mock button'));
    expect(onPress).toBeCalledTimes(1);
  });
});
