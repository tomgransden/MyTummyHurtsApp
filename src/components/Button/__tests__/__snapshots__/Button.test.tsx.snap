// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "mediumpurple",
      "borderColor": "mediumpurple",
      "borderRadius": 4,
      "borderWidth": 1,
      "marginBottom": 16,
      "opacity": 1,
      "padding": 8,
      "width": 160,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontFamily": "RubikBubbles-Regular",
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    My mock button
  </Text>
</View>
`;
